### PR TITLE
Initial validation

### PR DESCRIPTION
**Pull request by teneightfive**
This Pr adds basic step validation. 

It transforms the error object returned from Joi and sends the data back to the view. As well as that it sets the error messages for generic messages and custom messages for steps like _Already registered_.

It also adds the styles for the error summary component and updates some of the form element styles to match the prototype.

#### Multiple fields per page

screen shot 2017-03-16 at 14 43 21

#### Single field per page

screen shot 2017-03-16 at 14 43 43

### CODE REVIEW

**Comment by r4vi** [comment]
@teneightfive can you switch to this branch? https://github.com/nhsuk/register-with-a-gp-beta-web/tree/validation2

File: src/client/js/index.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
 const labelFocus = require('./modules/label-focus');
 const labelSelect = require('./modules/label-select');
+const errorSummary = require('./modules/error-summary');
```
Comment (r4vi) [inline src/client/js/index.js]: import?

File: src/server/plugins/register-form/steps/current-gp.js [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
+    .options({
+      language: {
+        any: { required: '!!Please tell us if you’re registered with a GP' },
+      },
+    }),
   'submit': Joi.any().optional().strip()
 });
 
 const title = 'Are you already registered with a GP?';
-const key = 'alreadyRegisteredWithGP';
```
Comment (r4vi) [inline src/server/plugins/register-form/steps/current-gp.js]: I think this is used somewhere else, so make sure to do find/replace

File: src/client/js/modules/error-summary.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/* If the current page shows errors, set the focus to the error messages */
+const $ = require('jquery');
```
Comment (r4vi) [inline src/client/js/modules/error-summary.js]: import?

File: src/server/plugins/register-form/steps/allergies.js [hunk 4]
```diff
@@ -24,7 +25,6 @@ const handlers = {
 export default {
   key,
   title,
-  fields,
```
Comment (r4vi) [inline src/server/plugins/register-form/steps/allergies.js]: can you update the jsdoc `@typedef` in `./index.js`

File: src/client/js/index.js (hunk 1 above)
Comment (teneightfive) [inline src/client/js/index.js]: Can come back to these but kept it using `require` for consistency.

File: src/client/js/index.js [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
 import title from './lib/title';
```
Comment (r4vi) [inline src/client/js/index.js]: can delete this

**Comment by Demotive** [comment]
Yeah. What @r4vi said. God, who hired you?

File: src/server/plugins/register-form/steps/current-gp.js (hunk 2 above)
Comment (teneightfive) [inline src/server/plugins/register-form/steps/current-gp.js]: 👍

File: src/server/plugins/register-form/steps/allergies.js (hunk 4 above)
Comment (teneightfive) [inline src/server/plugins/register-form/steps/allergies.js]: Done

File: src/client/js/index.js (hunk 5 above)
Comment (teneightfive) [inline src/client/js/index.js]: Will do a separate PR that tidies this and maybe the existing JS modules up.

Comment (r4vi) [inline src/client/js/index.js]: :+1:
